Sources/XCHammer/XcodeTarget.swift

```diff
@@ -531,6 +531,33 @@ public class XcodeTarget: Hashable, Equatable {
                 .map { "-D\($0)" }
         }
 
+        func extractSwiftVersion() -> String? {
+            // First look through the swiftcopts to see if `-swift-version <version>` has been specified
+            if let coptsArray = self.attributes[.swiftc_opts] as? [String] {
+                let versionOpt = coptsArray.filter { $0.hasPrefix("-swift-version") }.first
+                let version = versionOpt.map { $0.split(separator: " ") }?.last.map(String.init)
+                if version != nil {
+                    return version
+                }
+            }
+
+            // Second respect swift language version if explicitly defined
+            // This should be less common and i believe was only supported by old rules_apple rules
+            if let version = self.attributes[.swift_language_version] as? String {
+                return version
+            }
+
+            // Lastly if we have a swift dependency and can't determine the version, ask swiftc
+            if self.attributes[.has_swift_dependency] != nil || self.attributes[.has_swift_info] != nil {
+                // We don't have a guarantee the format will always be the same but it's unlikely to change
+                // ^^ I will regret these words (rmalik)
+                return try? ShellOut.shellOut(to: "xcrun swiftc -version | cut -d ' ' -f4")
```

Comment:
This returns the version followed by a blank line. Will a string with newlines work as a version?

Reply:
Good catch! I'll just trim this to be cautious